File: app/src/main/java/com/simplecity/amp_library/lyrics/LyricsView.java
package com.simplecity.amp_library.lyrics;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.simplecity.amp_library.model.Song;

interface LyricsView {

    void updateLyrics(@Nullable String lyrics);

    void showNoLyricsView(boolean show);

    void showQuickLyricInfoButton(boolean show);

    void showQuickLyricInfoDialog();

    void downloadQuickLyric();

    void launchQuickLyric(@NonNull Song song);
}

File: app/src/main/java/com/simplecity/amp_library/ui/screens/lyrics/LyricsView.kt
package com.simplecity.amp_library.ui.screens.lyrics

import com.simplecity.amp_library.model.Song

interface LyricsView {

    fun updateLyrics(lyrics: String?)

    fun showNoLyricsView(show: Boolean)

    fun showQuickLyricInfoButton(show: Boolean)

    fun showQuickLyricInfoDialog()

    fun downloadQuickLyric()

    fun launchQuickLyric(song: Song)
}